lib/index.js

```diff
@@ -21,7 +22,59 @@ module.exports.load = function load(rootFileName, options) {
 
     var loadedAbsoluteFileNames = []; // Do some housekeeping to avoid cycles
 
-    return (function readFileAndResolveIncludes(fileName) {
+    // Have to always keep public keys separate within resolvePublic
+    // If a private key is overwritten by a public key, it's an error, and vice-versa. Name your config keys something else!
+    // Empty object leaves and non-public keys are filtered out when requesting with public:true
+
+    function resolvePublic(obj, isToplevel, publicOnly) {
```

Comment:
Ah, withdrawn, you pass `true` and `false` below.

I'd still get rid of the `var publicOnly = ...` above, though.